spacy_llm/cache.py

```diff
@@ -5,8 +5,21 @@
 from spacy.tokens import Doc, DocBin
 from spacy.vocab import Vocab
 
+from .registry import registry
 
-class Cache:
+
+@registry.llm_misc("spacy.BatchCache.v1")
+def make_cache(
+    path: Optional[Union[str, Path]],
+    batch_size: int,
+    max_batches_in_mem: int,
+):
+    return BatchCache(
+        path=path, batch_size=batch_size, max_batches_in_mem=max_batches_in_mem
+    )
+
+
+class BatchCache:
```

Comment:
Is there a particular reason that we separate the factory from the class itself?

```suggestion
@registry.llm_misc("spacy.BatchCache.v1")
class BatchCache:
```

Reply:
> ...a policy that every factory should be a function and not a class could make sense for consistency's sake.

Yes, exactly that. There's no particular reason to do this here, it's to be consistent with the upcoming changes in https://github.com/explosion/spacy-llm/pull/81.